data/service/api/v1/summary.go

```diff
@@ -0,0 +1,113 @@
+package v1
+
+import (
+	"net/http"
+    "strconv"
+
+	//"github.com/tidepool-org/platform/data"
+	dataService "github.com/tidepool-org/platform/data/service"
+	//"github.com/tidepool-org/platform/page"
+	//"github.com/tidepool-org/platform/permission"
+	"github.com/tidepool-org/platform/request"
+)
+
+func SummaryRoutes() []dataService.Route {
+	return []dataService.Route{
+		dataService.MakeRoute("GET", "/v1/summary/:userId", Authenticate(GetSummary)),
+		dataService.MakeRoute("POST", "/v1/summary/:userId", Authenticate(UpdateSummary)),
+		dataService.MakeRoute("GET", "/v1/agedsummaries/:minutes", Authenticate(GetAgedSummaries)),
+	}
+}
+
+func GetSummary(dataServiceContext dataService.Context) {
+	res := dataServiceContext.Response()
+	req := dataServiceContext.Request()
+	dataClient := dataServiceContext.DataClient()
+
+	//details := request.DetailsFromContext(req.Context())
+	/*if details == nil {
+		request.MustNewResponder(res, req).Error(http.StatusUnauthorized, request.ErrorUnauthenticated())
+		return
+	}*/
+
+	responder := request.MustNewResponder(res, req)
+
+	id := req.PathParam("userId")
+	if id == "" {
+		responder.Error(http.StatusBadRequest, request.ErrorParameterMissing("id"))
+		return
+	}
+
+	summary, err := dataClient.GetSummary(req.Context(), id)
+	if err != nil {
+		responder.Error(http.StatusInternalServerError, err)
+		return
+	}
+
+	responder.Data(http.StatusOK, summary)
+}
+
+
+func UpdateSummary(dataServiceContext dataService.Context) {
+	res := dataServiceContext.Response()
+	req := dataServiceContext.Request()
+	dataClient := dataServiceContext.DataClient()
+
+	//details := request.DetailsFromContext(req.Context())
+	/*if details == nil {
+		request.MustNewResponder(res, req).Error(http.StatusUnauthorized, request.ErrorUnauthenticated())
+		return
+	}*/
+
+	responder := request.MustNewResponder(res, req)
+
+	id := req.PathParam("userId")
+	if id == "" {
+		responder.Error(http.StatusBadRequest, request.ErrorParameterMissing("id"))
+		return
+	}
+
+	summary, err := dataClient.UpdateSummary(req.Context(), id)
+	if err != nil {
+		responder.Error(http.StatusInternalServerError, err)
+		return
+	}
+
+	responder.Data(http.StatusOK, summary)
+}
+
+
+func GetAgedSummaries(dataServiceContext dataService.Context) {
+	res := dataServiceContext.Response()
+	req := dataServiceContext.Request()
+	dataClient := dataServiceContext.DataClient()
+
+	//details := request.DetailsFromContext(req.Context())
```

Comment:
I'm not sure I understand what is the use-case for this API?

It seems to return potentially unbounded number of user summaries - without regard to access permissions or user IDs (arbitrary set of user IDs? clinic patient user IDs? users who have shared with the session token owner?).

It should probably take pagination parameters (see `./data/service/api/v1/data_set.go` for an example).

Reply:
Added check for Service token for the permissions, as this API is only to be used by the task service to find what needs doing.
Will test pagination, need to make the task service handle pagination as well for this change.

Reply:
Decided against adding pagination to this for now, as writing custom pagination for this may be more confusing than its worth.